camlibs/ptp2/library.c

```diff
@@ -3623,7 +3623,7 @@ camera_capture_preview (Camera *camera, CameraFile *file, GPContext *context)
 		}
 
 		/* look for the JPEG SOI marker (0xFFD8) in data */
-		jpgStartPtr = (unsigned char*)memchr(jpgStartPtr, 0xff, size);
+		jpgStartPtr = (unsigned char*)memchr(jpgStartPtr, 0xff, ximage + size - jpgStartPtr);
```

Comment:
perhaps write the same expression
   size - (jpgStartPtr - ximage) 
to have valid pointer range arithmetic.

Reply:
I followed the existing examples in the same function as a guideline, and they all currently use it without parentheses (the one in the loop, the ones for `jpgEndPtr` etc).

If I went to change it, I'd rather track size in a separate variable but wanted to keep my change minimal.